### PR TITLE
Added onboarding and major changes to template

### LINKED ISSUE #1: [Suggestion] Adding a Spell Checker extension to recommended extensions list
There are some VS code extensions that are really good at spell checking. For example [Spell Right](https://marketplace.visualstudio.com/items?itemName=ban.spellright). 

I have some pros and cons regarding this.

### Pros 
- Having a spell checker would be helpful for users, especially if they are planning to publish their thoughts on the web. 
- Spell Checker uses the native spelling APIs which means the extension is pretty lightweight and also supports many languages.

### Cons
- It is an extra extension to install.
- ~~After installing the extension, the user will need to configure their Language in the`Spell Right Dictionary` using Command Palette. Until they do so, every word may have a red squiggle below it. I am not sure about this though.~~ It can be fixed by updating the `.vscode/settings.json`.

Screenshot 2020-06-25 at 5 50 39 PM

### LINKED ISSUE #27: Deprecate markdown links
Since Foam has it's own graph, we should deprecate the extension and start dogfooding our graph

### PR DESCRIPTION
**Pull request by riccardoferretti**
- cleaned up extension list, removed gitlens and prettier
- added onboarding instructions
  - including intro to vscode
  - including guide to keybindings for markdown editing (embarrassing to say I didn't know some of them!)
- adjusted some settings for better off-the-shelf experience (IMO)

Possibly contentious points:
- removing prettier
- "heavy" starter repo (I think it's totally fine, I put everything in one folder for easy removal)
- I kept the `prettier.singleQuote` setting, so that if someone installs prettier on their own, it will not cause the issue with the wikilink definitions

### CODE REVIEW

**Comment by jmg-duarte** [comment]
We can fix https://github.com/foambubble/foam-template/issues/1 by adding LTeX (it works for markdown too)
It provides helpful error messages as well.

Edit: Add LTeX example
Screenshot from 2021-01-22 11-06-31

**Comment by jmg-duarte** [comment]
Oh and also consider fixing this https://github.com/foambubble/foam-template/issues/27
Since we already have a graph

**Comment by riccardoferretti** [comment]
#27 is fixed by this PR

good point about #1 - I have added Spellright as the built-in spell checking extension, as it doesn't require the bootstrap needed by LTeX, still supports Latex, and it's lightweight as it leverages the OS spell checking engine.

**Comment by jmg-duarte** [comment]
LTeX bootstrap is automatic, VSCode takes care of it

José Duarte
https://jmg-duarte.github.io/

**Review by jmg-duarte — APPROVED** [review]

File: .vscode/extensions.json [hunk 1]
```diff
@@ -12,6 +12,9 @@
     "kortina.vscode-markdown-notes",
 
     // Image-pasting for markdown
-    "mushan.vscode-paste-image"
+    "mushan.vscode-paste-image",
+
+    // Spell checking for text, markdown and latex
+    "ban.spellright",
```
Comment (louiechristie) [inline .vscode/extensions.json:18]: @riccardoferretti why Spellright and not some other extension like [Code Spell Checker](https://marketplace.visualstudio.com/search?term=Code%20Spell%20Checker&target=VSCode&category=All%20categories&sortBy=Relevance)?

Code Spell Checker has 5.3 million downloads compared to Spellright's 300,000?

The reason I ask is I have this issue with Spellright https://github.com/bartosz-antosik/vscode-spellright/issues/210 when using it for other projects in other programming languages.

Comment (riccardoferretti) [reply .vscode/extensions.json:18]: This is just what I was using at the time, but per https://github.com/foambubble/foam/issues/1068 we have been thinking of removing it from the list of recommended extensions

Comment (louiechristie) [reply .vscode/extensions.json:18]: Thanks for the reply @riccardoferretti 👍

**Pull request merged**